pillar/top.sls.in

```diff
@@ -1,5 +1,33 @@
-metalk8s-@@VERSION:
+{%- set version = "@@VERSION" -%}
+{%- set roles = ['bootstrap', 'ca', 'etcd', 'master', 'node'] -%}
+
+# The mechanics of this file are very similar to those of `salt/top.sls(.in)`.
+# Please refer to the comments in that file for more background.
+
+{%- set version_match = "I@metalk8s:nodes:*:version:" ~ version -%}
+
+{%- macro role_match(name) -%}
+I@metalk8s:nodes:*:roles:{{ name }}
+{%- endmacro %}
+
+metalk8s-{{ version }}:
+  # The 'bootstrap' node requires the value of `metalk8s.iso_root_path`, which
+  # is only set in the `metalk8s.roles.bootstrap` Pillar. However, the
+  # compound-match below which assigns that Pillar to a node based on its role
+  # only works when the roles are available, i.e. late in the deployment (when
+  # the API server and associated `ext_pillar`s are in place).
+  # To work-around this, we assign the 'bootstrap' role Pillar data to all nodes
+  # for now, until either a better solution is found, or we get rid of
+  # `iso_root_path` and instead use what comes out of #924.
```

Comment:
We can also in the bootstrap script create a pillar file only for the current node assigning this bootstrap roles.

Have something like that in the top.sls
```
metalk8s-{{ version }}:
  '*':
    - {{ grains.id }}
   [...]
```
And then in our bootstrap script we just do 
```echo "{'metalk8s': {'nodes': {'$bootstrap_id': {'roles': ['ca', 'bootstrap', 'master', 'etcd']}}}"```
But then we have an issue with `ext_pillar_first` as this information in the static file will be take over the one in the apiserver...

So I don't really know.
Not really like the `ext_pillar_first` as the information from the apiserver should be taken over the one in the pillar file.

We should maybe get rid of "classic" pillar and only use ext_pillar to set all the information including the mine_functions... and so on.
And then in the master config `ext_pillar` we set the list in the right order we want them to get merged and then with "pillar" args from `def ext_pillar(minion_id, pillar)` we can get the information like roles and also the one from the command line.

Just tested `pillar` arg from ext_pillar contain all the merged informations of the previous ext_pillar + static pillar + **cmd line pillar** so for me it's something we can do

Reply:
> We can also in the bootstrap script create a pillar file only for the current node assigning this bootstrap roles.

Could even be simpler: pass it as a pillar value on the `salt-call` invocation in the script, like we do for several other pillar values.

Reply:
>  Not really like the ext_pillar_first as the information from the apiserver should be taken over the one in the pillar file.

Yes. And no. Not using `ext_pillar_first` simply doesn't work: we require the `ext_pillar` values to be able to dispatch on the roles (which are in the pillar...). Basically, when you don't use `ext_pillar_first`), you get crashes on the master side when it tries to compile the pillar top for a minion if said pillar top SLS contains matches on pillar values. So, no way around.

Reply:
> We should maybe get rid of "classic" pillar and only use ext_pillar to set all the information including the mine_functions... and so on.

Nah. If we can have static data, let's use static data.